nums/experimental/optimizer/graph.py

```diff
@@ -177,6 +178,7 @@ def copy(self, cluster_state, parent=None, new_ids=False):
         leaf._dtype = self._dtype
         leaf.parent = parent
         leaf.block = self.block
+        leaf.marker = self.marker
```

Comment:
Does this property have a purpose beyond its use in fusion? If not, lets add a comment here describing its use, or lets rename it to be less generic.